pkg/alertmanager/multitenant.go

```diff
@@ -479,10 +479,46 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 	userAM, ok := am.alertmanagers[userID]
 	am.alertmanagersMtx.Unlock()
 
-	if !ok || !userAM.IsActive() {
+	if !ok && am.fallbackConfig != "" {
+		// Give the user the fallback configuration and start its alertmanager
+		userAmConfig, err := amconfig.Load(am.fallbackConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to load the fallback configuration", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		transformedConfig, err := am.transformConfig(userID, userAmConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to transform the configuration", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		newAM, err := am.newAlertmanager(userID, transformedConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to start the Alertmanager", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		am.alertmanagersMtx.Lock()
+		am.alertmanagers[userID] = newAM
+		userAM = newAM
+		am.alertmanagersMtx.Unlock()
+
+		go func() {
```

Comment:
Have you considered doing it synchronously? I would expect it to take few hundred ms and, in my opinion, it's perfectly fine (even if it takes few seconds, given this is a one time thing).

Moreover, if we do it synchronously, from a logic perspective I would do (in order):
1. `am.store.SetAlertConfig()`
2. Update `am.alertmanagers`